external-crates/move/crates/move-bytecode-verifier/src/code_unit_verifier.rs

```diff
@@ -145,12 +145,35 @@ impl<'env> CodeUnitVerifier<'env, '_> {
         StackUsageVerifier::verify(verifier_config, self.module, &self.function_context, meter)?;
         type_safety::verify(self.module, &self.function_context, ability_cache, meter)?;
         locals_safety::verify(self.module, &self.function_context, ability_cache, meter)?;
-        reference_safety::verify(
+        let reference_safety_res = reference_safety::verify(
             verifier_config,
             self.module,
             &self.function_context,
             self.name_def_map,
             meter,
-        )
+        );
+        if verifier_config.sanity_check_with_regex_reference_safety {
+            let regex_res = regex_reference_safety::verify(
+                self.module,
+                &self.function_context,
+                &mut DummyMeter,
```

Comment:
Do we need to worry about this being "unmetered"? Would it be better to run this under a separate meter with a high bound just for safety sake?